types/fs-extra/index.d.ts

```diff
@@ -304,6 +307,27 @@ export interface ReadOptions {
     flag?: string;
 }
 
+export interface RemoveOptions {
```

Comment:
Do we need all those options to mimick the rimraf details?
It happens that rimarf has dependency on the `glob`:
https://github.com/isaacs/rimraf/blob/master/package.json
but I think we should not just use same options, could be enough to just implement interface from the `options` actual usage here:
https://github.com/jprichardson/node-fs-extra/blob/a571007f7066eb49e93343cd7f36bc373bc8dfc2/lib/remove/rimraf.js#L27

Reply:
I see.
But there's something I don't understand. Why is it that when I set a glob parameter, the function takes it into account?

Reply:
I've removed the glob part.
Is it correct now?

Reply:
yup, nearly there:
```ts
maxBusyTries?: number;
chmod?: typeof fs.chmod;
chmodSync?: typeof fs.chmodSync;
lstat?: typeof fs.lstat;
lstatSync?: typeof fs.lstatSync;
readdir?: typeof fs.readdir;
readdirSync?: typeof fs.readdirSync;
rmdir?: typeof fs.rmdir;
rmdirSync?: typeof fs.rmdirSync;
stat?: typeof fs.stat;
statSync?: typeof fs.statSync;
unlink?: typeof fs.unlink;
unlinkSync?: typeof fs.unlinkSync;
```
(to keep things together)

Reply:
I propose not to put the fs typing but rather the fs-extra ones.
When using the library, it would require to have fs AND fs-extra since the the types are incompatibles and `typeof fs.[anything]` is pretty hard to reproduce because it has a lot of properties like `_promisify_`.
So we should use `rmdir` instead of `fs.rmdir` in my opinion.
However fs-extra does not have sync versions of the functions.

Reply:
I think these should come from 'graceful-fs` (dependency):
https://github.com/jprichardson/node-fs-extra/blob/master/lib/remove/rimraf.js#L3
The import in the definition file is probably wrong.
What you'think?

Reply:
That would be the ideal.
But I face a recurrent issue in my projects. Consider the following ts file:
```typescript
import * as gfs from 'graceful-fs'
import fs from 'fs-extra'

interface test {
  unlink: typeof gfs.unlink
}

const obj: test = {
  unlink: fs.unlink
}
```
An error is thrown on the `unlink` property of obj:
```
Property '__promisify__' is missing in type '{ (path: string | Buffer, callback: (err: ErrnoException) => void): void; (path: string | Buffer): Promise<void>; }' but required in type 'typeof unlink'.ts(2741)
fs.d.ts(726, 18): '__promisify__' is declared here.
test.ts(5, 3): The expected type comes from property 'unlink' which is declared here on type 'test'
```
The types of graceful-fs solve this issue whereas with graceful-fs or fs it's almost impossible to provide a clean function.

Reply:
@oganexon @peterblazejewicz Is this ready to merge?

Reply:
I think so but @peterblazejewicz doesn't seem to reply.
I've summarized the situation here:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43329#issuecomment-604549428